server/api/impl/upstream/uphttp/upstream.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package uphttp
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/api/impl/upstream"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+type Upstream struct {
+	rootURL   string
+	appSecret string
+}
+
+func NewUpstream(app *api.App) *Upstream {
+	return &Upstream{app.Manifest.HTTPRootURL, app.Secret}
+}
+
+func (u *Upstream) InvokeNotification(n *api.Notification) error {
+	// TODO
+	resp, err := u.post("", u.rootURL+"/notify/"+string(n.Subject), n)
```

Comment:
TODO: Need to change notification path to a relative URL, provided by the caller in a `Subscription`; maybe re-consider using the `Call` format for notifications in general, just a different `Call.Type`? Filed https://mattermost.atlassian.net/browse/MM-31125